core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -148,6 +244,16 @@
      */
     public abstract long lastModified() throws IOException;
 
+    /**
+     * Gets the file’s Unix mode, if meaningful.
+     * @return for example, 0644 ~ {@code rw-r--r--}; -1 by default, meaning unknown or inapplicable
+     * @throws IOException if checking the mode failed
+     * @since FIXME
+     */
+    public int mode() throws IOException {
```

Comment:
CheckReturnValue?

Reply:
Not sure why. No other similar methods have that annotation. And the method obviously has no side effects, so it seems unlikely someone would accidentally call it as a statement.